### ISSUE TITLE
Issue with File.binread when passing FakeFS::Pathname object

### ISSUE THREAD

**Issue by tovodeverett** [fakefs/fakefs#326]
I'm still investigating, but I'm seeing an issue using `fakefs` when I pass a `FakeFS::Pathname` object into `File.binread`.  I suspect the issue also shows up with `#read`.  It appears that `FakeFS::Pathname` doesn't respond to the `[]` method in https://github.com/defunkt/fakefs/blob/master/lib/fakefs/globber.rb#L7.

An immediate work around that comes to mind is coercing the `FakeFS::Pathname` object into a string in `FakeFS::Globber.expand`, but it's probably better to add that behavior to `FakeFS::Pathname`.

I'll report back with what I find.

**Comment by tovodeverett** [comment]
Given that `Pathname` doesn't respond to `[]`, it appears that `FakeFS::Globber` should coerce paths to strings before using `[]` on them!  I'll try to work on a proper patch for this (including tests).  A comprehensive test suite for this would ensure that all entry points to `fakefs` that accept paths work with either `String` or `Pathname` objects.

In the interim, my monkey patch is to wrap the public method (`expand`, `path_components`, and `regexp`) in `FakeFS::Globber` to prepend `pattern = pattern.to_s` to each method.  Add the following to your `spec_helper.rb`:

```
module FakeFS
  module Globber
    [:expand, :path_components, :regexp].each do |method_sym|
      old_method = instance_method(method_sym)
      define_method(method_sym) do |pattern|
        old_method.bind(self).(pattern.to_s)
      end
    end
  end
end
```

**Comment by tovodeverett** [comment]
See https://github.com/fakefs/fakefs/pull/330 for a resolution.

**Comment by tovodeverett** [comment]
Superseded by #336.

**Comment by tovodeverett** [comment]
Resolved by merge of #336 in 0.9.0.
